File: python/해시/의상.py

""" 레벨 2. 의상 """

def solution(clothes):
    
    all_clothes = {}

    for _, key in clothes:
        all_clothes[key] = all_clothes.setdefault(key, 0) + 1

    answer = 1

    for value in all_clothes.values():
        answer *= (value+1)

    return answer-1

print(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]))     # 5
print(solution([["crow_mask", "face"], ["blue_sunglasses", "face"], ["smoky_makeup", "face"]]))                 # 3

File: Java/src/자료구조/의상.java
package 자료구조;

import java.util.HashMap;

public class 의상 {

    public static void main(String[] args) {

        String[][] c1 = {{"yellow_hat", "headgear"}, {"blue_sunglasses", "eyewear"}, {"green_turban", "headgear"}};
        System.out.println(solution(c1));

        String[][] c2 = {{"crow_mask", "face"}, {"blue_sunglasses", "face"}, {"smoky_makeup", "face"}};
        System.out.println(solution(c2));
    }   

    public static int solution(String[][] clothes) {

        HashMap<String, Integer> map = new HashMap<>();

        for (String[] c : clothes) {
            map.put(c[1], map.getOrDefault(c[1], 0) + 1);
        }

        int answer = 1;
        for (int v : map.values()) {
            answer *= (v + 1);
        }

        return answer - 1;
    }
}
